public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts

```diff
@@ -0,0 +1,43 @@
+import { parse } from 'yaml';
+import { KubeConfig } from '../Kubernetes.types';
+
+export const onKubeConfigValueChange = (
+  value: string | undefined,
+  updateFormMutator: (configValue: string | undefined, nameValue: string | undefined) => void
+) => {
+  const defaultName = getClusterNameFromKubeConfig(value);
+  updateFormMutator(value, defaultName);
+};
+
+const getClusterNameFromKubeConfig = (value: string | undefined): string | undefined => {
+  if (value) {
+    try {
+      const parsedYAML = parse(value);
+      const clusters: KubeConfig[] = parsedYAML?.clusters;
+      if (clusters && clusters.length) {
+        const clusterWithName = clusters.find((item: any) => item?.name != null);
```

Comment:
I see you added typings, please don't forget to remove that `any` from find.

Reply:
sorry, lost it, I'm going to check all any right now